src/utils/aeternity.js

```diff
@@ -14,18 +15,20 @@ import { IS_MOBILE_DEVICE } from './index';
 let sdk;
 let contractV1;
 let contractV2;
+let contractV3;
 
 const initTippingContractIfNeeded = async () => {
   if (!sdk) throw new Error('Init sdk first');
-  if (!contractV1) {
-    contractV1 = await sdk.getContractInstance(TIPPING_V1_INTERFACE, {
-      contractAddress: process.env.VUE_APP_CONTRACT_V1_ADDRESS,
-    });
-  }
-  if (!contractV2 && process.env.VUE_APP_CONTRACT_V2_ADDRESS) {
-    contractV2 = await sdk.getContractInstance(TIPPING_V2_INTERFACE, {
-      contractAddress: process.env.VUE_APP_CONTRACT_V2_ADDRESS,
-    });
+  if (!contractV1
+     || (!contractV2 && process.env.VUE_APP_CONTRACT_V2_ADDRESS)
+     || (!contractV3 && process.env.VUE_APP_CONTRACT_V3_ADDRESS)) {
```

Comment:
This condition is unclear. As I see it should be called only once, so proposing to define near a flag `isTippingContractsInitialised`, switch it to `true` if we have tried to initialize them, and to check that flag here instead of this complex condition.

Reply:
I would suggest addressing these suggestions in another PR.

They are not blocking the feature to be merged. Let's be leaner.